src/dc_context.c

```diff
@@ -307,6 +339,49 @@ char* dc_get_blobdir(const dc_context_t* context)
  * INI-handling, Information
  ******************************************************************************/
 
+static int is_settable_config_key(const char* key)
+{
+	for (int i = 0; i < str_array_len(config_keys); i++) {
+		if (strcmp(key, config_keys[i]) == 0) {
+			return 1;
+		}
+	}
+	return 0;
+}
+
+
+static int is_gettable_config_key(const char* key)
+{
+	for (int i = 0; i < str_array_len(sys_config_keys); i++) {
+		if (strcmp(key, sys_config_keys[i]) == 0) {
+			return 1;
+		}
+	}
+	return is_settable_config_key(key);
+}
+
+
+static char* get_config_keys_str()
+{
+	dc_strbuilder_t  ret;
+	dc_strbuilder_init(&ret, 0);
+
+	for (int i = 0; i < str_array_len(config_keys); i++) {
+		if (ret.buf != NULL) {
```

Comment:
it must read `ret.buf[0]!=0` or `strlen(ret.buf)>0` as `dc_strbuilder_init()` always sets the pointer to an empty string.
the condition  is always `ret.buf != NULL` is always true therefore.

sorry for the bad or missing internal documentation.

Reply:
oh, my bad. I actually saw this, but somehow immediately forgot even after you mentioned the possible solution in the comments. will fix in a couple of minutes.

Reply:
Done